src/coreclr/jit/emitarm64.cpp

```diff
@@ -11821,11 +11825,18 @@ void emitter::emitDispImm(ssize_t imm, bool addComma, bool alwaysHex /* =false *
 
         if ((imm & 0xFFFFFFFF00000000LL) != 0)
         {
-            printf("0x%llx", imm);
+            if (isAddrOffset)
+            {
+                printf("%llXH", imm);
+            }
+            else
+            {
+                printf("0x%llx", imm);
+            }
         }
         else
         {
-            printf("0x%02x", (unsigned)imm);
+            printf("%02XH", (unsigned)imm);
```

Comment:
There is also preexisting UB here. I think we should be using `%z` format specifier for these printf's taking `ssize_t` (e.g. `%02zXH` -- but may want to double check).

Reply:
Ah sorry, I have not noticed that you have added the `unsigned` cast now. Then it should be ok.